File: Easy/uva_11185/uva_11185.py
while True:
    try:
        n = int(input())
    except:
        break
    
    if n < 0:
        break

    ans = ''
    if n == 0:
        ans = '0'
    
    while n > 0:
        ans = str(n % 3) + ans
        n //= 3

        '''
        n = 10 ans = '1'
        n = 3 ans = '10'
        n = 1 ans = '101'
        n = 0 out of while loop
        '''
    print(ans)

File: Easy/uva_11185/uva_11185.cpp
// UVA 11185 - Ternary
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    long long n;
    while ( (cin >> n) ) {
        if (n < 0) break;
        if (n == 0) { cout << "0\n"; continue; }
        string ans;
        while (n > 0) {
            ans.push_back(char('0' + (n % 3)));
            n /= 3;
        }
        reverse(ans.begin(), ans.end());
        cout << ans << '\n';
    }
    return 0;
}
